### ISSUE TITLE
Super Game Boy palette/border support?

### ISSUE THREAD

**Issue by bryc** [libretro/gambatte-libretro#79]
A number of GBx emulators support basic features of the SGB like custom palettes and borders. To name a few, VBA-M, KiGB, bgb, Gameyob (DS) and Lameboy (DS). Unfortunately the VBA-M libretro ports are GBA-only and don't seem to have any plans to support classic GBx, which is a shame. In fact there are no libretro cores that have this feature.

I honestly just want the custom palettes - Lots of DMG games have special palettes for SGB that add nicer colors than just B&W, such as Kirby 2 and Pokemon Blue. 

Could it be possible to support those SGB color modes for games that use it? Would it be accepted as a PR? Or would this functionality be better set in a hypothetical VBAM-GBC core. The GB experience just isn't complete without it.

What do you think?

Here is a comparison:

![image](https://cloud.githubusercontent.com/assets/1408749/23345447/623cc8c0-fc5c-11e6-94b0-eb3c755e26b2.png)

Left: Kirby 2 using GBC palette
Right: Kirby 2 using SGB palette

**Comment by hizzlekizzle** [comment]
Yeah, SGB palettes would be great, and we would likely accept non-hacky PRs to VBAM, beetle-gba or Gambatte that provided it. Currently, the only way to get SGB functionality is via bsnes, but it's weird, only works through the command line and is broken everywhere but Windows for some unknown reason.

**Comment by sergiobenrocha2** [comment]
This issue was moved to libretro/libretro-meta#49

**Comment by SuperMeatBro** [comment]
But would it be possible to implement?

**Comment by paradadf** [comment]
@bryc I believe, after https://github.com/libretro/gambatte-libretro/pull/113 this issue can be closed.

**Comment by bryc** [comment]
#113 is not the same thing as that pull request. Those are just SGB palette templates, not the game-controlled palettes.

When switching to 'sgb' setting, I don't get the 3 colors in the intro, I just get this:

![image](https://user-images.githubusercontent.com/1408749/47041081-e0476200-d155-11e8-83a4-3c69bb279928.png)

**Comment by paradadf** [comment]
@bryc ok, then @hizzlekizzle should re-open the issue. Sorry about the misunderstanding.

**Comment by Cee123** [comment]
I noticed exactly the same thing the other day. The colours are NOT the accurate colouring of Super Game Boy enhanced games (such as Pokemon Red/Blue) and will only display in one colour. There are also no support for Super Game Boy borders. I guess our only alternative is to use mGBA until someone decides to actually implement these features.

**Comment by Cee123** [comment]
Guys, I added a bounty for SGB features to be added, if anyone is interested.
https://www.bountysource.com/issues/76534488-incorrect-coloring-on-super-gameboy-enhanced-games-and-super-gameboy-borders
